v2/tools/generator/internal/functions/property_assignment_function_test.go

```diff
@@ -375,3 +376,58 @@ func TestGolden_PropertyAssignmentFunction_WhenTypeRenamed(t *testing.T) {
 	// and in the name of the Assign*() functions.
 	test.AssertSingleTypeDefinitionGeneratesExpectedCode(t, "PropertyTypeRenamed", receiverDefinition)
 }
+
+func TestGolden_PropertyAssignmentFunction_WhenSharedObjectVersion(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+	idFactory := astmodel.NewIdentifierFactory()
+	injector := astmodel.NewFunctionInjector()
+
+	location2020 := test.CreateObjectDefinition(
+		test.Pkg2020,
+		"Location",
+		test.FullAddressProperty)
+
+	location2021 := test.CreateObjectDefinition(
+		test.Pkg2021,
+		"Location",
+		test.FullAddressProperty)
+
+	location2022 := test.CreateObjectDefinition(
+		test.Pkg2022,
+		"Location",
+		test.FullAddressProperty)
+
+	person2020 := test.CreateObjectDefinition(
+		test.Pkg2020,
+		"Person",
+		astmodel.NewPropertyDefinition("Residence", "residence", location2020.Name()))
+
+	person2022 := test.CreateObjectDefinition(
+		test.Pkg2022,
+		"Person",
+		astmodel.NewPropertyDefinition("Residence", "residence", location2022.Name()))
+
+	definitions := make(astmodel.TypeDefinitionSet)
+	definitions.AddAll(location2020, location2021, location2022)
+	definitions.AddAll(person2020, person2022)
+
+	cfg := config.NewObjectModelConfiguration()
+	builder := storage.NewConversionGraphBuilder(cfg, "v")
+	builder.Add(test.Pkg2020, test.Pkg2021, test.Pkg2022)
+	graph, err := builder.Build()
+	g.Expect(err).To(BeNil())
+
+	conversionContext := conversions.NewPropertyConversionContext(definitions, idFactory).WithConversionGraph(graph)
+
+	assignFrom, err := NewPropertyAssignmentFunction(person2020, person2022, conversionContext, conversions.ConvertFrom)
```

Comment:
Would it make sense to create `NewPropertyAssignmentFunction` for `Location` as well here?